### PR TITLE
Decompile a few objects

### PR DESCRIPTION
**Pull request by Cleverking2003**

### CODE REVIEW

File: src/lever_wall.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        break;
+    }
+    if (r6 != 0) {
+        sb = arg0;
+        x = arg0->base.x >> 12;
+        y = arg0->base.y >> 12;
+        sub_08001408(arg0->base.unk56, sub_080025AC(arg0->base.unk56, x, y), 0, 0);
+        sub_0800E0E4(arg0, x, y);
+        PlaySfx(&arg0->base, 0xff << 1);
```
Comment (jiangzhengwenjz) [inline src/lever_wall.c:30]: `510`

File: src/lever_wall.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#include "functions.h"
+#include "lever_wall.h"
+
+static void LeverWallAction(struct Object2*);
+static void LeverWallSetFunc(struct Object2*);
+static void nullsub_115(struct Object2*);
+
+void LeverWallAction(struct Object2* arg0) {
+    u16 r6 = 0, x, y;
```
Comment (jiangzhengwenjz) [inline src/lever_wall.c:9]: Does it match with
```C
u16 r6 = 0;
s16 x, y;
```
?

File: src/lever_wall.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    struct Object2 *sb;
+    switch (arg0->object->subtype1) {
+    case 0:
+        if (sub_080029BC(gCurLevelInfo[arg0->base.unk56].unk65E) == arg0->object->unkF) {
+            r6 = 1;
+        }
+        break;
+    case 1:
+        if (sub_080029E4(gCurLevelInfo[arg0->base.unk56].unk65E) == arg0->object->unkF) {
+            r6 = 1;
```
Comment (jiangzhengwenjz) [inline src/lever_wall.c:20]: `r6` is `bool16`. Use `TRUE`/`FALSE`

File: src/cupie.c [hunk 4]
```diff
@@ -0,0 +1,879 @@
...
+    {0x0038, 0xff40, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0x0020, 0xff20, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffe0, 0xff20, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffc8, 0xff40, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffa0, 0xffa0, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffa0, 0x0060, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffc8, 0x00c0, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0xffe0, 0x00e0, 0x0000, 0x0000, 0x10, 0x00, 0x00, 0x00},
+    {0x0000, 0x0000, 0x0000, 0x0000, 0x00, 0x00, 0x00, 0x00},
+};
```
Comment (jiangzhengwenjz) [inline src/cupie.c:103]: The first 2 fields of `struct Unk_08353510` are `s16` so some entries here are negative.

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: src/cupie.c [hunk 5]
```diff
@@ -0,0 +1,879 @@
...
+                    arg0->base.xspeed = 0x100;
+                }
+                else if (arg0->base.xspeed < -0x100) {
+                    arg0->base.xspeed = -0x100;
+                }
+            }
+        }
+        else {
+            if (arg0->base.xspeed < 0) {
+                if ((arg0->base.xspeed += 4) << 0x10 > 0) {
```
Comment (jiangzhengwenjz) [inline src/cupie.c:485]: Add a comment here since this is a compiler trick that can invoke defined behavior

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: src/cupie.c [hunk 6]
```diff
@@ -0,0 +1,879 @@
...
+    obj->unk1E = 0;
+    obj->unk20 = 0;
+    obj->unk11 = 0;
+    obj->unk12 = 0;
+    obj->unk14 = 0;
+    obj->unk16 = 0;
+    obj->unk18 = 0;
+    obj2_2 = CreateObject(arg0->base.unk56, obj);
+    obj2_2->base.parent = arg0;
+    obj2 = obj2_2;
```
[the remark this answers is not in the document]
Comment (Cleverking2003) [reply src/cupie.c:660]: Yes, that one's needed for matching

**Review by jiangzhengwenjz — APPROVED** [review]

**Pull request merged**